### PR TITLE
metanorma/pngcheck#4 add release-tag workflow

### PR DESCRIPTION
**Pull request by CAMOBAP**
Related PR https://github.com/metanorma/pngcheck/pull/6

### CODE REVIEW

File: .github/workflows/release-tag.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    types: [ metanorma/pngcheck ]
+
+jobs:
+  prepare-ieee:
+    uses: ./.github/workflows/prepare.yml
+    secrets:
+      pat_token: ${{ secrets.METANORMA_CI_PAT_TOKEN }}
+
+  tag_repo:
+    runs-on: ubuntu-18.04
```
Comment (ronaldtse) [inline .github/workflows/release-tag.yml]: Can we use ubuntu-latest?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: .github/workflows/release-tag.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: release-tag
+
+on:
+  repository_dispatch:
+    types: [ metanorma/pngcheck ]
+
+jobs:
+  prepare-ieee:
+    uses: ./.github/workflows/prepare.yml
```
Comment (maxirmx) [inline .github/workflows/release-tag.yml]: How does it get the file  ```./.github/workflows/prepare.yml``` ?

File: .github/workflows/release-tag.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: release-tag
+
+on:
+  repository_dispatch:
+    types: [ metanorma/pngcheck ]
+
+jobs:
+  prepare-ieee:
```
Comment (maxirmx) [inline .github/workflows/release-tag.yml]: ```prepare-ieee``` job name looks misleading here. Can we change it ?

**Review by maxirmx — CHANGES_REQUESTED** [review]
- It looks like this workflow releases ```pngcheck-ruby``` after a trigger from  ```pngcheck``` but without any tests of Ruby code. If I understand it correctly it is too risky

- My approach to ```zlib``` does not look robust enough.  https://github.com/metanorma/pngcheck-ruby/issues/30
Reverse merge from main shall help as temporary solution for this PR

File: .github/workflows/release-tag.yml (hunk 2 above)
Comment (CAMOBAP) [reply .github/workflows/release-tag.yml]: Fixed

File: .github/workflows/release-tag.yml (hunk 3 above)
Comment (CAMOBAP) [reply .github/workflows/release-tag.yml]: Right, fixed

File: .github/workflows/release-tag.yml (hunk 1 above)
Comment (CAMOBAP) [reply .github/workflows/release-tag.yml]: Fixed

**Comment by CAMOBAP** [comment]
> * It looks like this workflow releases `pngcheck-ruby` after a trigger from  `pngcheck` but without any tests of Ruby code. If I understand it correctly it is too risky

The whole sequence is:
`pngcheck` -> `release-tag` -> `test-and-release`

So we will do pre-release testing. To me it looks really close to what we are doing manually

> * My approach to `zlib` does not look robust enough.  #30
>   Reverse merge from main shall help as temporary solution for this PR

:+

**Comment by CAMOBAP** [comment]
@maxirmx I just rebased this branch from latest main, did I missed something?

File: .github/workflows/release-tag.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+on:
+  repository_dispatch:
+    types: [ metanorma/pngcheck ]
+
+jobs:
+  tag_repo:
+    runs-on: ubuntu-latest
+    if: startsWith(github.event.client_payload.ref, 'refs/tags/metanorma-v')
+    needs: [ prepare-ieee ]
```
Comment (maxirmx) [inline .github/workflows/release-tag.yml]: I guess it shall be removed

**Review by maxirmx — APPROVED** [review]
- pngcheck is not maintained and does not include any CI scripts or tests. 
- All tests that we have are within pngcheck-ruby project/repo. 
- Also pngcheck-ruby is not a simple wrapper, it has its own logic.

So, the likely scenario for me is:
1. make a branch
2. update pngcheck version
3. update pngcheck-ruby logic
4. implement/adjust tests 
5. merge
6. tag/release

I can imagine scenario where some pngcheck update is pushed to pngcheck-ruby and the latter is released automatically but it is highly unlikely case for me.  However, if it considered more important for automation let's do it :)

**Comment by CAMOBAP** [comment]
@maxirmx thanks for sharing feedback, to be honest I thought that this repo is a really similar wrapper as we use on `mn2pdf+mn2pdf-ruby` and `mnconvert+mnconvert-ruby` and so on

Anyway let's how it will work (or not)

**Comment by ronaldtse** [comment]
@CAMOBAP this gem is a little bit different from mn2pdf and mnconvert because it is not a simple wrapper, as @maxirmx mentioned, but it is good to share the same workflow, as long as there is a way to insert changes / prevent releases.

**Comment by CAMOBAP** [comment]
> @CAMOBAP this gem is a little bit different from mn2pdf and mnconvert because it is not a simple wrapper, as @maxirmx mentioned, but it is good to share the same workflow, as long as there is a way to insert changes / prevent releases.

Just to be on the same page. Current solution will try to release pngcheck-ruby one pngcheck released. By updating version/url/sha256. If tests on pngcheck-ruby will passed, CI will do gem release.

So there is no way to "insert changes" between `pngcheck` release and `pngcheck-ruby`.

It looks to me that there is no potential to automate releases more than they are already automated

**Comment by ronaldtse** [comment]
That's fine. The point is if a new pngcheck causes pngcheck-ruby to fail, then the release will fail. This is when we need to do manual fixes. The proposed way of automating is good.

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thank you @CAMOBAP !

**Pull request merged**